packages/bff/src/services/toolsService.ts

```diff
@@ -0,0 +1,393 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+
+import {
+  ApiKey,
+  ClientAssertion,
+  ConsumerKey,
+  FailedValidation,
+  SuccessfulValidation,
+  validateClientKindAndPlatformState,
+  validateRequestParameters,
+  verifyClientAssertion,
+  verifyClientAssertionSignature,
+} from "pagopa-interop-client-assertion-validation";
+import {
+  AgreementId,
+  ApiError,
+  ClientId,
+  EServiceId,
+  ItemState,
+  PurposeId,
+  TenantId,
+  unsafeBrandId,
+} from "pagopa-interop-models";
+import { getAllFromPaginated, WithLogger } from "pagopa-interop-commons";
+import {
+  agreementApi,
+  authorizationApi,
+  bffApi,
+  catalogApi,
+  purposeApi,
+} from "pagopa-interop-api-clients";
+import {
+  AgreementProcessClient,
+  CatalogProcessClient,
+  PagoPAInteropBeClients,
+} from "../providers/clientProvider.js";
+import { BffAppContext } from "../utilities/context.js";
+import {
+  activeAgreementByEserviceAndConsumerNotFound,
+  clientAssertionPublicKeyNotFound,
+  ErrorCodes,
+  eserviceDescriptorNotFound,
+  missingActivePurposeVersion,
+  multipleAgreementForEserviceAndConsumer,
+  purposeIdNotFoundInClientAssertion,
+} from "../model/domain/errors.js";
+import { TokenGenerationValidationStepFailure } from "../../../api-clients/dist/bffApi.js";
+
+export function toolsServiceBuilder(clients: PagoPAInteropBeClients) {
+  return {
+    async validateTokenGeneration(
+      clientId: string | undefined,
+      clientAssertion: string,
+      _clientAssertionType: string,
+      _grantType: string,
+      ctx: WithLogger<BffAppContext>
+    ): Promise<bffApi.TokenGenerationValidationResult> {
```

Comment:
Even if not present in Scala I would add a log here, something like `Validating token generation for client ${clientId}` - in all other BFF services we have a log at the beginning of the handler function